From e419b05d3ec7a4758c14e5653f11a23be4d162ff Mon Sep 17 00:00:00 2001
From: Stephan Bergmann <sbergman@redhat.com>
Date: Thu, 8 May 2014 13:10:46 +0200
Subject: [PATCH] Resolves: rhbz#1092589 Thoroughly check whether JRE is still
 present

...not only on Mac OS X.  Was able to reproduce this on Fedora 20 where current
JRE was /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.5.x86_64 but for whatever reason
there was also a left-behind /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.0.x86_64 tree
(containing just a handful of sub-dirs, but no real content) that was still
recorded in my ~/.config/libreoffice/4/user/config/javasettings_Linux_X86_64.xml

Change-Id: Ie477c5a506a430f6c29525f6c558dbc18bbf1c48
(cherry picked from commit f5ed2f4e926016d23617355c94dd2292b40e6986)
---
 jvmfwk/plugins/sunmajor/pluginlib/sunjavaplugin.cxx | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/jvmfwk/plugins/sunmajor/pluginlib/sunjavaplugin.cxx b/jvmfwk/plugins/sunmajor/pluginlib/sunjavaplugin.cxx
index a3449c3..d3650f9 100644
--- a/jvmfwk/plugins/sunmajor/pluginlib/sunjavaplugin.cxx
+++ b/jvmfwk/plugins/sunmajor/pluginlib/sunjavaplugin.cxx
@@ -765,10 +765,10 @@ javaPluginError jfw_plugin_existJRE(const JavaInfo *pInfo, sal_Bool *exist)
     {
         ret = JFW_PLUGIN_E_ERROR;
     }
-#ifdef MACOSX
     //We can have the situation that the JavaVM runtime library is not
     //contained within JAVA_HOME. Then the check for JAVA_HOME would return
     //true although the runtime library may not be loadable.
+    //Or the JAVA_HOME directory of a deinstalled JRE left behind.
     if (ret == JFW_PLUGIN_E_NONE && *exist == sal_True)
     {
         OUString sRuntimeLib = getRuntimeLib(pInfo->arVendorData);
@@ -796,7 +796,6 @@ javaPluginError jfw_plugin_existJRE(const JavaInfo *pInfo, sal_Bool *exist)
                        + sRuntimeLib + " \n");
         }
     }
-#endif
     return ret;
 }
 
-- 
1.9.0

